cloud/blockstore/vhost-server/backend_rdma.cpp

```diff
@@ -22,19 +29,102 @@
 
 using namespace NCloud::NBlockStore;
+using namespace NThreading;
 
 namespace {
 
 ////////////////////////////////////////////////////////////////////////////////
 
+constexpr ui32 REQUEST_TIMEOUT_MSEC = 86400000;
+
+////////////////////////////////////////////////////////////////////////////////
+
+class TStorageDataClient final
+    : public IBlockStore
+{
+private:
+    const IStoragePtr Storage;
+
+public:
+    TStorageDataClient(IStoragePtr storage)
+        : Storage(std::move(storage))
+    {}
+
+    void Start() override
+    {}
+
+    void Stop() override
+    {}
+
+    TStorageBuffer AllocateBuffer(size_t bytesCount) override
+    {
+        return Storage->AllocateBuffer(bytesCount);
+    }
+
+#define BLOCKSTORE_DONT_IMPLEMENT_METHOD(name, ...)                            \
+    TFuture<NProto::T##name##Response> name(                                   \
+        TCallContextPtr callContext,                                           \
+        std::shared_ptr<NProto::T##name##Request> request) override            \
+    {                                                                          \
+        Y_UNUSED(callContext);                                                 \
+        Y_UNUSED(request);                                                     \
+        const auto& type = GetBlockStoreRequestName(EBlockStoreRequest::name); \
+        return MakeFuture<NProto::T##name##Response>(TErrorResponse(           \
+            E_NOT_IMPLEMENTED,                                                 \
+            TStringBuilder() << "Unsupported request " << type.Quote()));      \
+    }                                                                          \
+
+#define BLOCKSTORE_IMPLEMENT_METHOD(name, ...)                                 \
+    TFuture<NProto::T##name##Response> name(                                   \
+        TCallContextPtr callContext,                                           \
+        std::shared_ptr<NProto::T##name##Request> request) override            \
+    {                                                                          \
+        return Storage->name(std::move(callContext), std::move(request));      \
+    }                                                                          \
+
+    BLOCKSTORE_GRPC_SERVICE(BLOCKSTORE_DONT_IMPLEMENT_METHOD)
+    BLOCKSTORE_LOCAL_SERVICE(BLOCKSTORE_IMPLEMENT_METHOD)
+
+#undef BLOCKSTORE_IMPLEMENT_METHOD
+#undef BLOCKSTORE_DONT_IMPLEMENT_METHOD
+};
+
+////////////////////////////////////////////////////////////////////////////////
+
+class TRetryPolicy : public NClient::IRetryPolicy
+{
+private:
+    NClient::IRetryPolicyPtr Impl;
+
+public:
+    TRetryPolicy(NClient::TClientAppConfigPtr config)
+        : Impl(NClient::CreateRetryPolicy(std::move(config)))
+    {}
+
+    NClient::TRetrySpec ShouldRetry(
+        NClient::TRetryState& state,
+        const NProto::TError& error) override
+    {
+        if (error.GetCode() == E_BS_INVALID_SESSION) {
```

Comment:
Why do we treat session error as retriable?

Reply:
I saw this error coming sometimes when we restart disk agent
In the regular nrd based volume flow path this error is handled by remount of the volume and retry https://github.com/ydb-platform/nbs/blob/65c25012afcc9da5fa0b3decf3d1fd1d188992f6/cloud/blockstore/libs/client/session.cpp#L947

With the vhost-rdma based data path we rely on the rdma partition actor which is running but have no io's going through it to remount the volume after some timeout and then the request will be executed correctly